From b4feb6f3f019a06d118009b795d0f568d77da3b7 Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Wed, 3 Jul 2024 17:47:29 +0200
Subject: [PATCH 646/954] feat(stm32mp21-fdts): add SD card support for
 stm32mp215f-dk board

Enable SDMMC1 node for stm32mp215f-dk board.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 99e80290bf732497fdd55a2b4b7e415d3859b385 feat(stm32mp21-fdts): add SD card support for stm32mp215f-dk board]
Change-Id: I22c75222b81f66c0077fdac6fc0a9d8b8740aa4d
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/391263
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
---
 fdts/stm32mp215f-dk.dts | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/fdts/stm32mp215f-dk.dts b/fdts/stm32mp215f-dk.dts
index d0abd4d963..082218da0b 100644
--- a/fdts/stm32mp215f-dk.dts
+++ b/fdts/stm32mp215f-dk.dts
@@ -178,6 +178,16 @@
 	status = "okay";
 };
 
+&sdmmc1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&sdmmc1_b4_pins_a>;
+	st,neg-edge;
+	bus-width = <4>;
+	vmmc-supply = <&vdd_flash>;
+	vqmmc-supply = <&vddio1>;
+	status = "okay";
+};
+
 &usart2 {
 	pinctrl-names = "default", "idle", "sleep";
 	pinctrl-0 = <&usart2_pins_a>;
-- 
2.53.0

